spec.md

```diff
@@ -792,133 +727,44 @@ In the following example, all variants have the type `UInt<0>`{.firrtl}.
-See [@sec:probes] for details.
-
-Using probe-type ports, modules may expose internals for reading and forcing without routing wires out of the design.
+## Probe Types
 
-This is often useful for testing and verification, where probe types allow reads of the entities to be explicitly exported without hard-coding their place in the design.
-Instead, by using probe-type references, a testbench module may express accesses to the internals which will resolve to the appropriate target language construct by the compiler (e.g., hierarchical reference).
+Probe types expose and provide access to circuit components contained inside a module.
 
-Reference ports are not expected to be synthesizable or representable in the target language and are omitted in the compiled design; they only exist at the FIRRTL level.
+Probe types are useful for testing and verification, since they allow a design to read to and write from circuit components without knowing the hierarchical path to that component in the design.
 
-Reference-type ports are statically routed through the design using the `define`{.firrtl} statement.
-
-There are two reference types, `Probe`{.firrtl} and `RWProbe`{.firrtl}, described below.
-These are used for indirect access to probes of the data underlying circuit constructs they originate from, captured using `probe`{.firrtl} expressions (see [@sec:probes]).
-
-`Probe`{.firrtl} types are read-only, and `RWProbe`{.firrtl} may be used with `force`{.firrtl} and related statements.
-Prefer the former as much as possible, as read-only probes impose fewer limitations and are more amenable to optimization.
-
-Probe references must always be able to be statically traced to their target, or to an external module's output reference.
-This means no conditional connections via sub-accesses, multiplexers, or other means.
-
-Reference types compose with optional groups (see [@sec:optional-groups].
-A reference type may be associated with an optional group.
-When associated with an optional group, the reference type may only be driven from that optional group.
-
-### Probe Types
-
-Probe types are reference types used to access circuit elements' data remotely.
-
-There are two probe types: `Probe`{.firrtl} and `RWProbe`{.firrtl}.
-`RWProbe`{.firrtl} is a `Probe`{.firrtl} type, but not the other way around.
-
-Probe types are parametric over the type of data that they refer to, which is always passive (as defined in [@sec:passive-types]) even when the probed target is not (see [@sec:probes-and-passive-types]).
-Probe types cannot contain reference types.
-
-Conceptually probe types are single-direction views of the probed data-flow point.
-They are references to the data accessed with the probe expression generating the reference.
+There are two probe types, `Probe<T>`{.firrtl} is a read-only variant and `RWProbe<T>`{.firrtl} is a read-write variant.
+In both cases, `T`{.firrtl} must be a ground type (see [@sec:ground-types]).
```

Comment:
is this an intentional change to the spec? Because the example you have deleted is not a ground type. If you are trying to make a functional change to the spec that probes have to be ground types, please make a different PR for that.

Reply:
I thought this is how these are being implemented from Chisel now. Probes of aggregates don't work well with the the Port Lowering ABI v1 (i.e., `LowerTypes`) on public modules. Hence, this is a restriction that I think is already in place.

We're at a `[major]` spec boundary given that the spec already has public modules. So, I'm fine if this is just a `[major]` change.

Reply:
This is the kind of thing I'm looking to flush out with this spec work.

I'm fine with making this a `[major]` change. But let's be sure not to release a new version until the dust has all settled.

Reply:
sure i am not objecting to a spec change, just seems like that should be done broken out from a big rewrite of things, given that its like a one sentence change and deletion of an example to say that probes don't work on aggregates.

I am not making a statement on how chisel is implementing them now/ near future, defer to @dtzSiFive on that.

Reply:
@mwachs5 I promoted the PR to `[major]`.

Reply:
I'm flipping this back to `[patch]` and removing this line.

This point represents a pretty big gap. But let me leave this without the restriction, and we can address it after all the low-hanging fruit is gone.

Reply:
thanks :) nit I see you reverted the sentence but you didn't revert the example, which is now just sort of ambiguous and confusing 😂 If you think this is a noncontroversial change can just open a PR right now to remove it, no need to block on this larger (in lines of text) change